src/Core/Types.Filters/Extensions/FilterObjectFieldDescriptorExtensions.cs

```diff
@@ -7,18 +7,45 @@ namespace HotChocolate.Types.Filters
 {
     public static class FilterObjectFieldDescriptorExtensions
     {
+        public static IObjectFieldDescriptor<T> UseFilter<T>(
```

Comment:
I also added a new UseFilter here that now is able to infer the type from the property.